src/app/components/location-detail/location-detail.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Character } from 'src/app/models/character';
+import { Location } from 'src/app/models/location';
+import { ApiService } from 'src/app/services/api.service';
+
+@Component({
+  selector: 'app-location-detail',
+  templateUrl: './location-detail.component.html',
+  styleUrls: ['./location-detail.component.css','./../../app.component.css']
+})
+export class LocationDetailComponent implements OnInit {
+	public location: Location;
+  public residents: Character[] = [];
+
+	constructor(
+		private readonly dataService: ApiService,
+		private readonly route: ActivatedRoute,
+    private readonly router: Router
+  ) { }
+
+	ngOnInit() {
+		let characterId = this.route.snapshot.params["id"];
+		this.dataService.getLocation(characterId).subscribe(location => {
```

Comment:
you have basically the same logic in another place, I believe it could be unified somehow